worker/src/test/scala/org/apache/celeborn/service/deploy/cluster/PushDataWithChecksumSuite.scala

```diff
@@ -0,0 +1,120 @@
+import org.apache.celeborn.common.protocol.message.StatusCode
+import org.apache.celeborn.common.unsafe.Platform
+import org.apache.celeborn.common.util.{PushDataHeaderUtils, Utils}
+import org.apache.celeborn.service.deploy.MiniClusterFeature
+
+class PushDataWithChecksumSuite extends AnyFunSuite
+  with Logging with MiniClusterFeature with BeforeAndAfterAll {
+  var masterPort = 19097
+
+  override def beforeAll(): Unit = {
+    logInfo("test initialized , setup Celeborn mini cluster")
+    val (master, _) = setupMiniClusterWithRandomPorts()
+    masterPort = master.conf.masterPort
+  }
+
+  override def afterAll(): Unit = {
+    logInfo("all test complete , stop Celeborn mini cluster")
+    shutdownMiniCluster()
+  }
+
+  test("push data and checksum error") {
+    val APP = s"app-${System.currentTimeMillis()}"
+    val SHUFFLE_ID = 0
+    val MAP_ID = 0
+    val ATTEMPT_ID = 0
+    val MAP_NUM = 1
+    val PARTITION_NUM = 3
+
+    val clientConf = new CelebornConf()
+      .set(CelebornConf.MASTER_ENDPOINTS.key, s"localhost:$masterPort")
+    val lifecycleManager = new LifecycleManager(APP, clientConf)
+    val shuffleClient = new ShuffleClientImpl(APP, clientConf, UserIdentifier("mock", "mock"))
+    shuffleClient.setupLifecycleManagerRef(lifecycleManager.self)
+
+    // ping and reserveSlots
+    val DATA0 = RandomStringUtils.secure().next(10).getBytes(StandardCharsets.UTF_8)
+    shuffleClient.pushData(
+      SHUFFLE_ID,
+      MAP_ID,
+      ATTEMPT_ID,
+      0,
+      DATA0,
+      0,
+      DATA0.length,
+      MAP_NUM,
+      PARTITION_NUM)
+
+    val partitionLocationMap =
+      shuffleClient.getPartitionLocation(SHUFFLE_ID, MAP_NUM, PARTITION_NUM)
+    val location = partitionLocationMap.get(SHUFFLE_ID)
+    val shuffleKey = Utils.makeShuffleKey(APP, SHUFFLE_ID)
+
+    val batchId = 123 // mock batch id
+    val bodyLength = 100
+    val buffer: Array[Byte] = Array.fill(PushDataHeaderUtils.BATCH_HEADER_SIZE)(
+      0.toByte) ++ RandomStringUtils.secure().next(bodyLength).getBytes(StandardCharsets.UTF_8)
+    PushDataHeaderUtils.buildDataHeader(buffer, MAP_ID, ATTEMPT_ID, batchId, bodyLength)
+    val checksum = PushDataHeaderUtils.computeHeaderChecksum32(buffer)
+    Platform.putInt(buffer, PushDataHeaderUtils.CHECKSUM_OFFSET, checksum + 1) // error checksum
+
+    val nettyBuffer = new NettyManagedBuffer(Unpooled.wrappedBuffer(buffer));
+    val client =
+      shuffleClient.getDataClientFactory.createClient(location.getHost, location.getPushPort)
+    val pushData = new PushData(
+      PartitionLocation.Mode.PRIMARY.mode,
+      shuffleKey,
+      location.getUniqueId,
+      nettyBuffer)
+    val callback = new RpcResponseCallback() {
+      override def onSuccess(response: ByteBuffer): Unit = {
+        assert(false, "push data should fail")
+      }
+
+      override def onFailure(e: Throwable): Unit = {
+        if (e.getMessage == StatusCode.PUSH_DATA_CHECKSUM_FAIL.toString) {
```

Comment:
assert(e.getMessage == StatusCode.PUSH_DATA_CHECKSUM_FAIL.toString, "push data should fail for checksum error, but get " + e.getMessage)

Reply:
Thank you, fixed.